packages/router/utils.ts

```diff
@@ -861,9 +863,7 @@ export function getPathContributingMatches<
 >(matches: T[]) {
   return matches.filter(
     (match, index) =>
-      index === 0 ||
-      (!match.route.index &&
-        match.pathnameBase !== matches[index - 1].pathnameBase)
+      index === 0 || (match.route.path && match.route.path.length > 0)
```

Comment:
2 issues fixed here:

* Looking at `route.index` doesn't work since index routes can have paths
* Using `pathnameBase` doesn't work for nested splat routes since they don't add to `pathnameBase`, so instead we just look for the route do define a non-empty `path` value (also this means this becomes a function that we can use directly in remix since remix doesn't expose `pathnameBase` on matches)